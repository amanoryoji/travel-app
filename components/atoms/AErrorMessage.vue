<script lang="ts">
type FontSize = "lg" | "md" | "sm";

export default {
  props: {
    message: {
      type: String,
    },
    fontSize: {
      type: String as PropType<FontSize>,
      default: () => "md",
    },
  },
  setup() {
    return {};
  },
};
</script>

<template>
  <p
    v-if="message"
    class="el_errorMessage"
    :class="[
      { el_errorMessage__hidden: !message },
      `el_errorMessage__${fontSize}`,
    ]"
  >
    {{ message }}
  </p>
</template>

<style lang="scss" scoped>
.el_errorMessage {
  height: 1rem;
  font-size: 1rem;
  color: $color-warning;

  &__sm {
    height: 0.75rem;
    font-size: 0.75rem;
  }
  &__lg {
    height: 1.25rem;
    font-size: 1.25rem;
  }

  &__hidden {
    visibility: hidden;
  }
}
</style>
