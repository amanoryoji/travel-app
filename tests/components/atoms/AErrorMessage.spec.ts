import { VueWrapper, shallowMount } from "@vue/test-utils";

import AErrorMessage from "/components/atoms/AErrorMessage.vue";


describe("ErrorMessage", () => {
  let wrapper: VueWrapper<AErrorMessage>;
  beforeEach(() => {
    wrapper = shallowMount(AErrorMessage);
  });

  it("コンポーネントがマウントされる", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  describe("スナップショット", () => {
    it("デフォルト値", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it("全propを指定", async () => {
      await wrapper.setProps({
        message: "myMessag",
        fontSize: "lg",
      });
      expect(wrapper.html()).toMatchSnapshot();
    });

    describe("fontSize", () => {
      it("sm", async () => {
        await wrapper.setProps({ fontSize: "sm" });
        expect(wrapper.html()).toMatchSnapshot();
      });

      it("md", async () => {
        await wrapper.setProps({ fontSize: "md" });
        expect(wrapper.html()).toMatchSnapshot();
      });

      it("lg", async () => {
        await wrapper.setProps({ fontSize: "lg" });
        expect(wrapper.html()).toMatchSnapshot();
      });
    });
  });
});
