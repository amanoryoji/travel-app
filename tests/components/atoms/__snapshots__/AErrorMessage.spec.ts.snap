// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage > スナップショット > fontSize > lg 1`] = `"<!--v-if-->"`;

exports[`ErrorMessage > スナップショット > fontSize > md 1`] = `"<!--v-if-->"`;

exports[`ErrorMessage > スナップショット > fontSize > sm 1`] = `"<!--v-if-->"`;

exports[`ErrorMessage > スナップショット > デフォルト値 1`] = `"<!--v-if-->"`;

exports[`ErrorMessage > スナップショット > 全propを指定 1`] = `"<p data-v-06d1af55=\\"\\" class=\\"el_errorMessage el_errorMessage__lg\\">myMessag</p>"`;
